var/spack/repos/builtin/packages/metall/package.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Metall(CMakePackage):
+    """An allocator for persistent memory"""
+
+    homepage = "https://github.com/LLNL/metall"
+    git      = "git@github.com:LLNL/metall.git"
+
+    maintainers = ['KIwabuchi', 'rogerpearce', 'mayagokhale']
+
+    version('develop', branch='develop')
+
+    variant('boost', default=False, description='Install a proper version of Boost')
+    depends_on('boost@1.64:', type=('link'), when='+boost')
+
+    def cmake_args(self):
```

Comment:
Is there a CMake flag to enable/disable boost support?

Reply:
Actually, I'd like to consult you about how to use Spack for a header-only library properly.
Metall is a header-only library that requires some header files in Boost (build is not needed for Boost too). 
So, you don't need Boost to install Metall using Spack because all you have to do is just copying the header files of Metall. 
Of course, users still need a proper version of Boost to build Metall with their applications. That's why I added '+boost' option so that users don't have to download an appropriate version of Boost manually.
Thank you,

Reply:
If the `boost` headers are always required, I would remove the variant and always depend on boost. Spack packages should never rely on system installations.

Reply:
Thank you for the suggestion. I'll modify it that way.